### ISSUE TITLE
Add fields to Issue : Item , Serial No

### ISSUE THREAD

**Issue by johnwslongland** [frappe/erpnext#43559]
**Is your feature request related to a problem? Please describe.**
My feature request does not relate to a problem

**Describe the solution you'd like**
Please may two fields be added to the "Details" section on "Issue".
Item and Serial No
These fields should be non-mandatory.

**Describe alternatives you've considered**
No alternatives considered

**Additional context**
Many times when a customer raises an  issue, an item is involved. And some items may have a serial no.
The only way to capture this currently, is in the text-box.

**Comment by OmelMedina** [comment]
@johnwslongland 

I just played with this. I made a custom item field and custom serial number field in the Issue Doctype. Then you select an Item which might have a serial number. If the Item does have a serial number, it will show you a serial number field with relevant serial numbers to the item only. If the Item doesn't have a serial number, you will get a notification on the bottom right side saying "No Serial Numbers found for the selected Item". I wrote a custom script for the client side and selected Issue as the Doctype. Both fields aren't mandatory. Is it close to what you were looking for?

**Comment by johnwslongland** [comment]
@Glitch-Trap-1 
That's correct ! Nice touch .... the "show_alert" pop-up. I didn't think of that !!!

**Comment by OmelMedina** [comment]
@johnwslongland 

Alright here are the steps

Step 1: Navigate to  **Issue** and click on **Customize Form**

Step 2: Create two **Link** Fields named **Item** and **Serial Number** under the details section. (double check that custom field names are **custom_item** and **custom_serial_number** respectively or else the script won't work)

Step 3: Click on the **Item** field and scroll down to **Options** on the right side of your screen. Type in **Item** as that is the DocType you are referencing for that field. 

Step 4: Click on the **Serial Number** field and scroll down to **Options** on the right side of your screen. Type in **Serial No** as that is the DocType you are referencing for that field. 

Step 5: Save the customized form and go to the home page

Step 6: Use the search bar on the top right and type in **Client Script List** and hit **enter**

Step 7: Click on the **Add Client Script** button

Step 8: Add your custom script name **(ex. Issue SN Script)**

Step 9: Select your DocType in this case which is **Issue**. Leave **Apply To** as **Form**

Step 10: Paste this custom script code into **Script** field

```
frappe.ui.form.on('Issue', {
    refresh: function(frm) {
        frm.set_query("custom_serial_number", function(doc) {
            return {
                filters: {
                    'item_code': doc.custom_item
                }
            };
        });
    },
    
    custom_item: function(frm) {
        frm.set_value('custom_serial_number', ''); // Clear the custom_serial_number field when item changes
        
        if (frm.doc.custom_item) {
            frappe.call({
                method: 'frappe.client.get_list',
                args: {
                    doctype: 'Serial No',
                    filters: {
                        'item_code': frm.doc.custom_item
                    },
                    fields: ['name'],
                    limit: 1
                },
                callback: function(response) {
                    if (response.message && response.message.length > 0) {
                        frm.set_df_property('custom_serial_number', 'hidden', 0); // Show the field
                        frm.refresh_field('custom_serial_number');
                    } else {
                        frm.set_df_property('custom_serial_number', 'hidden', 1); // Hide the field
                        frappe.show_alert({
                            message: __('No Serial Numbers found for the selected Item.'),
                            indicator: 'orange'
                        }, 5);
                    }
                }
            });
        } else {
            frm.set_df_property('custom_serial_number', 'hidden', 1); // Hide the field when no item is selected
        }
        frm.refresh_field('custom_serial_number');
    }
});
```

Step 11: Tick the **Enabled** box and save the script

Step 12: Click on your profile and click on **Reload** (This will clear cache) and refresh the page for good measure

Now it should filter serial numbers based on the item chosen. Make sure you create the serial numbers for the Item beforehand. Funny enough I only did this because I felt as it would be useful for the company I work at. If you need more help on how to implement this. Just let me know

**Comment by johnwslongland** [comment]
Nice stuff @Glitch-Trap-1 
I will look at your code and add to the code I have. 
And as you say, it is usefull in your company as well. I am hoping that the developers also agree that it is a usefull thing to have on an "Issue" doctype and that they add this to the core-code.

Surely, there are cases that an issue is related to an Item ? I agree, dont make it manadatory, because it can be argued that not all issues are related to items, but at least have the fields available if there are users that have use-cases that can use  it.

Great effort ! Thank you !

**Comment by OmelMedina** [comment]
@johnwslongland 

I do agree with that. It might not pertain to all businesses but they should have no reason not to implement this into the core logic as a function that can be enabled. Most products we sell have serial numbers and we like to record the serial number of the machine that is having issues. Tracking issues becomes much simpler if I can just type in the Serial Number of the specific product since one company might have multiple machines. Integrating the Item and Serial Number doctypes into Issues shouldn't be difficult but I do understand that having more features means a wider scope to manage. 

Hopefully it helps you and good luck!
